monitoring/voltage-monitoring/src/main/java/com/farao_community/farao/monitoring/voltage_monitoring/json/VoltageMonitoringResultSerializer.java

```diff
@@ -35,20 +40,48 @@ public void serialize(VoltageMonitoringResult voltageMonitoringResult, JsonGener
         jsonGenerator.writeStartObject();
 
         jsonGenerator.writeStringField(TYPE, VOLTAGE_MONITORING_RESULT);
+        jsonGenerator.writeStringField(STATUS, voltageMonitoringResult.getStatus().toString());
+        // VOLTAGE_VALUE
         jsonGenerator.writeArrayFieldStart(VOLTAGE_VALUES);
+        serializeExtremeVoltageValue(voltageMonitoringResult, jsonGenerator);
+        jsonGenerator.writeEndArray();
+        // APPLIED_RAS
+        jsonGenerator.writeArrayFieldStart(APPLIED_RAS);
+        serializeAppliedRas(voltageMonitoringResult, jsonGenerator);
+        jsonGenerator.writeEndArray();
+
+        jsonGenerator.writeEndObject();
+    }
+
+    private void serializeExtremeVoltageValue(VoltageMonitoringResult voltageMonitoringResult, JsonGenerator jsonGenerator) throws IOException {
         for (Map.Entry<VoltageCnec, ExtremeVoltageValues> entry :
-            voltageMonitoringResult.getExtremeVoltageValues().entrySet()
-                .stream().sorted(Comparator.comparing(e -> e.getKey().getId()))
-                .collect(Collectors.toList())) {
+                voltageMonitoringResult.getExtremeVoltageValues().entrySet()
+                        .stream().sorted(Comparator.comparing(e -> e.getKey().getId()))
+                        .collect(Collectors.toList())) {
             jsonGenerator.writeStartObject();
             jsonGenerator.writeStringField(CNEC_ID, entry.getKey().getId());
             jsonGenerator.writeNumberField(MIN, entry.getValue().getMin());
             jsonGenerator.writeNumberField(MAX, entry.getValue().getMax());
             jsonGenerator.writeEndObject();
         }
-        jsonGenerator.writeEndArray();
+    }
 
-        jsonGenerator.writeEndObject();
+    private void serializeAppliedRas(VoltageMonitoringResult voltageMonitoringResult, JsonGenerator jsonGenerator) throws IOException {
+        for (Map.Entry<State, Set<NetworkAction>> entry : voltageMonitoringResult.getAppliedRas().entrySet().stream().sorted(Comparator.comparing(e -> e.getKey().getId()))
+                .collect(Collectors.toList())) {
+            jsonGenerator.writeStartObject();
+            jsonGenerator.writeStringField(INSTANT, entry.getKey().getInstant().toString());
+            Optional<Contingency> optContingency = entry.getKey().getContingency();
+            if (optContingency.isPresent()) {
+                jsonGenerator.writeStringField(CONTINGENCY, optContingency.get().getId());
+            }
+            jsonGenerator.writeArrayFieldStart(REMEDIAL_ACTIONS);
+            for (NetworkAction networkAction : entry.getValue().stream().sorted(Comparator.comparing(NetworkAction::getId)).collect(Collectors.toList())) {
+                jsonGenerator.writeString(networkAction.getId());
+            }
```

Comment:
```suggestion
            entry.getValue().stream().map(NetworkAction::getId).sorted().foreach(jsonGenerator::writeString);
```

Reply:
This would cause an error: incompatible thrown types java.io.IOException in functional expression